control/src/benchmarking.rs

```diff
@@ -1,15 +1,108 @@
-//! Benchmarking setup for gamedao-control
-use super::*;
+#![cfg(feature = "runtime-benchmarks")]
+
+ use crate::*;
+ use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite, whitelisted_caller};
+ use frame_system::RawOrigin;
+ use sp_runtime::{DispatchError, traits::{Saturating}};
+ use sp_std::vec;
+
+/// Fund accounts with tokens, needed for org interactions
+fn fund_accounts<T: Config>(account_ids: &Vec<T::AccountId>) -> Result<(), DispatchError> {
+	let multiplier: u32 = 2;
+	let amount = T::MinimumDeposit::get().saturating_mul(multiplier.into());
+	for account_id in account_ids {
+		T::Currency::deposit(T::ProtocolTokenId::get(), account_id, amount)?;
+		T::Currency::deposit(T::PaymentTokenId::get(), account_id, amount)?;
+	}
+	Ok(())
+}
 
-#[allow(unused)]
-use crate::Pallet as Control;
-use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite};
-use frame_system::RawOrigin;
-use sp_std::vec;
 
 benchmarks! {
 
-	simple_benchmark {}: {}
+	create_org {
+		let caller: T::AccountId = whitelisted_caller();
+		fund_accounts::<T>(&vec![caller.clone()])?;
+		let org_nonce = Nonce::<T>::get();
+	}: 	_(
+		RawOrigin::Signed(caller.clone()),
+		caller.clone().into(),
+		(0..255).collect(),
+		(0..255).collect(),
+		OrgType::Individual,
+		AccessModel::Open,
+		FeeModel::NoFees,
+		T::Balance::default(),
+		T::CurrencyId::default(),
+		T::CurrencyId::default(),
+		100,
+		None
+	)
+	verify {
+		assert!(OrgByNonce::<T>::get(org_nonce).is_some());
+	}
+
+	disable_org {
+		let caller: T::AccountId = whitelisted_caller();
+		fund_accounts::<T>(&vec![caller.clone()])?;
+		let org_id = <Pallet::<T> as ControlBenchmarkingTrait<T::AccountId, T::Hash>>::create_org(caller.clone()).unwrap();
+	}: _(RawOrigin::Root, org_id)
+	verify {
+		assert!(OrgState::<T>::get(org_id) == ControlState::Inactive);
+	}
+
+	enable_org {
+		let caller: T::AccountId = whitelisted_caller();
+		fund_accounts::<T>(&vec![caller.clone()])?;
+		let org_id = <Pallet::<T> as ControlBenchmarkingTrait<T::AccountId, T::Hash>>::create_org(caller.clone()).unwrap();
+	}: _(RawOrigin::Root, org_id)
+	verify {
+		assert!(OrgState::<T>::get(org_id) == ControlState::Active);
```

Comment:
This check doesn't actually ensure that the execution of the benchmark was successful since it already has an `Active` state.